int32 detector_type
int32 detector_mode

int32 NO_DETECTOR                   = 0
int32 DEMO_DETECTOR                 = 1
int32 OBSTACLE_DETECTOR             = 2

int32 DEFAULT_MODE                  = 0
